### PR TITLE
Jjrunner

### PR DESCRIPTION
**Pull request by nastasi-oq**
First working version with some lack (see issues).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+~~~~
+$ # an example of calls used to build locally and upload
+$ # to an experimental launchpad repository all the engine packages stack
+$ jjrunner.py -a '{"GEM_MASTER_BRANCH": "jjrunner", "GEM_SET_DEBUG": "true", "build_for_ubuntu_trusty": "false", "build_for_ubuntu_xenial": "true"}' master_oq-python
+
+$ jjrunner.py -a '{"GEM_MASTER_BRANCH": "jjrunner", "GEM_SET_DEBUG": "true", "build_for_ubuntu_trusty": "false", "build_for_ubuntu_xenial": "true", "JOB_NAME": "master_oq-libs" }' zdevel_oq-libs
+
+$ jjrunner.py -a '{"GEM_TEST_FOR_JJRUNNER": "true", "GEM_JENKINS_REASON": "Started by timer", "GEM_LAUNCHPAD_REPO": "test4", "GEM_MASTER_BRANCH": "jjrunner", "GEM_SET_DEBUG": "true", "run_dev_tests": "false", "build_for_ubuntu_trusty": "true" }' master_oq-engine
+~~~~
```
Comment (daniviga) [inline README.md]: You are mixing the `rst` syntax within an `md` file. Even if GitHub is smart enough please use `markdown` for consistency with other repos.

File: jjrunner.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    params['JENKINS_HOME'] = {'name': 'JENKINS_HOME', 'defa': expanduser("~"),
+                              'desc': 'auto-generated'}
+    params['BUILD_NUMBER'] = {'name': 'BUILD_NUMBER', 'defa': 1,
+                              'desc': 'auto-generated'}
+    try:
+        g = git.cmd.Git(os.getcwd())
+        params['GIT_BRANCH'] = {'name': 'GIT_BRANCH',
+                                'defa': g.rev_parse('--abbrev-ref', 'HEAD'),
+                                'desc': 'auto-generated'}
+    except:
```
Comment (daniviga) [inline jjrunner.py:90]: Wouldn't be better to trap specific exceptions?

File: jjrunner.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        f_com.close()
+        os.chmod(f_com_name, stat.S_IREAD | stat.S_IEXEC | stat.S_IWUSR)
+
+        f_main = open(f_main_name, "w")
+        f_main.write('#!/bin/bash\n. %s\n%s\n' %
+                     (f_args_name, f_com_name))
+        f_main.close()
+        os.chmod(f_main_name, stat.S_IREAD | stat.S_IEXEC | stat.S_IWUSR)
+
+        # import ipdb ; ipdb.set_trace()
```
Comment (daniviga) [inline jjrunner.py]: Please remove this

**Review by daniviga — APPROVED** [review]
LGTM but I cannot test it because Jenkins jobs are very OS dependent and they depends also on some helpers that are available only in Jenkins (like lxc wrappers).

Maybe it would be good to document also these dependencies (like sudo configurations...)

MN: add a new issue about it

File: jjrunner.py (hunk 2 above)
Comment (nastasi-oq) [reply jjrunner.py:90]: retrieve GIT_BRANCH value is optional; so, if something goes wrong, the fallback is always print the WARNING and go ahead.

**Pull request merged**